go/cli/mcap/cmd/filter.go

```diff
@@ -240,16 +240,25 @@ func filter(
 			if err != nil {
 				return err
 			}
+			// if any topics match an includeTopic, add it.
 			for _, matcher := range opts.includeTopics {
 				if matcher.MatchString(channel.Topic) {
 					channels[channel.ID] = markableChannel{channel, false}
 				}
 			}
-			for _, matcher := range opts.excludeTopics {
-				if !matcher.MatchString(channel.Topic) {
+			// if a topic does not match any excludeTopic, add it.
+			if len(opts.excludeTopics) != 0 {
+				shouldInclude := true
+				for _, matcher := range opts.excludeTopics {
+					if matcher.MatchString(channel.Topic) {
+						shouldInclude = false
+					}
+				}
+				if shouldInclude {
 					channels[channel.ID] = markableChannel{channel, false}
 				}
 			}
```

Comment:
break?

Wonder if the outer if is necessary – seems like this could be structured as "if no exclude topics match, add the channel"

Reply:
we're in nested for-loop territory, breaks can get confusing and don't seem neccessary here.
The outer if can't be trivially removed, otherwise if the user specifies no excludeTopics but they do specify includeTopics, all topics will get included no matter what. I could change the condition to `if len(opts.includeTopics) == 0` but not sure if that'd reduce complexity or increase readability.